feat(react-native): add push notification documentation according to the new SDK changes (#712)
GetStream/stream-video-js: packages/react-native-sdk/docusaurus/docs/reactnative/05-advanced/02-push-notifications.mdx

// highlight-end
```

#### Initialize on app launch

> We need to configure the Firebase library, setup the callkeep library and register VoIP at the app launch. To do this, add the following methods to your existing `didFinishLaunchingWithOptions` method,

Reviewer: ```suggestion
We need to configure the Firebase library, set up the callkeep library, and register VoIP at the app launch. To do this, add the following methods to your existing `didFinishLaunchingWithOptions` method,
```

}
```

## Setup the push config for the SDK

The SDK automatically processes the incoming push notifications once the setup above is done if the push config has been set using `StreamVideoRN.setPushConfig`. To do this follow the steps below,
### Add the ability to statically navigate to screens in your app

When a user taps on the push notification and the JS engine is not ready, they should still be able to navigate to the screen that shows the active call. You can achieve this by adding the ability to [navigate without the navigation property in the react-navigation library](https://reactnavigation.org/docs/navigating-without-navigation-prop/).

The following is an example implementation of a utility file that has helpers to statically navigate in the app:

```ts title="src/utils/staticNavigation.ts"
import { createNavigationContainerRef } from '@react-navigation/native';

import { RootStackParamList } from '../navigation/types';

export const navigationRef = createNavigationContainerRef<RootStackParamList>();

/**
 * This is used to run the navigation logic from root level even before the navigation is ready
 */

> export const StaticNavigate = (

Reviewer 2: Change everywhere, so will use camelCase of PascalCase ;)
```suggestion
export const staticNavigate = (
```

</NavigationContainer>
```

### Setup the push config

Once we have setup the methods to navigate the app from a static method we are ready to call the `StreamVideoRN.setPushConfig` method. Below is an example of how this method can be called,

```ts title="src/utils/setPushConfig.ts"
import {
  StreamVideoClient,
  StreamVideoRN,
} from '@stream-io/video-react-native-sdk';
import { AndroidImportance } from '@notifee/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { STREAM_API_KEY } from '../../constants';
import { StaticNavigate } from './staticNavigationUtils';

> export function setPushConfig() {

Reviewer 2: Please break long comment lines so it'll be easier to read.

like 310 321 etc...

  StreamVideoRN.setPushConfig({
    ios: {
      // add your push_provider_name for iOS that you have setup in Stream dashboard
      pushProviderName: __DEV__ ? 'apn-video-staging' : 'apn-video-production',
    },
    android: {
      // add your push_provider_name for Android that you have setup in Stream dashboard
      pushProviderName: __DEV__ ? 'firebase-video-staging' : 'firebase-video-production',
      // configure the notification channel to be used for incoming calls for Android.
      incomingCallChannel: {
        id: 'stream_incoming_call',
        name: 'Incoming call notifications',

>         importance: AndroidImportance.HIGH,

Reviewer 2: ```suggestion
        // This is the advised importance of receiving incoming call notifications.
        // This will ensure that the notification will be on top of application
        importance: AndroidImportance.HIGH,
```

</NavigationContainer>
```

### Setup the push config

> Once we have setup the methods to navigate the app from a static method we are ready to call the `StreamVideoRN.setPushConfig` method. Below is an example of how this method can be called,

Reviewer: ```suggestion
Once we have set up the methods to navigate the app from a static method we are ready to call the `StreamVideoRN.setPushConfig` method. Below is an example of how this method can be called,
```

    navigateAcceptCall: () => {
      staticNavigate({ name: 'ActiveCallScreen', params: undefined });
    },
    // add the callback to be executed when a notification is tapped,
    // but the user did not press accept or decline, used for navigation
    navigateToIncomingCall: () => {
      staticNavigate({ name: 'IncomingCallScreen', params: undefined });
    },
    // add the async callback to create a video client 
    // for incoming calls in the background on a push notification
    createStreamVideoClient: async () => {
      // note that since the method is async, 
      // you can call your server to get the user data or token or retrieve from offline storage.
      const userId = await AsyncStorage.getItem('@userId');
      const userName = await AsyncStorage.getItem('@userName');
      const token = await AsyncStorage.getItem('@userToken');
      if (!username || !userToken) return undefined;
      const user = { id: userId, name: userName, token };
      return new StreamVideoClient({
        apiKey: STREAM_API_KEY, // pass your stream api key
        user,
        token,
      })
    },
  });
};
```

> Now call the method outside of your application cycle. That is, alongside your `AppRegistry.registerComponent()` method call at the entry point of your application code. This is because the app can be opened from a dead state through a push notification and in that case, we need to use the config as soon as the JS bridge is initialized. Following is an example,

Reviewer: ```suggestion
Now, call the method outside of your application cycle. That is, alongside your `AppRegistry.registerComponent()` method call at the entry point of your application code. This is because the app can be opened from a dead state through a push notification and in that case, we need to use the config as soon as the JS bridge is initialized. Following is an example,
```
